pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -675,3 +678,46 @@ type DigitalOceanSDConfig struct {
 	// +optional
 	RefreshInterval *v1.Duration `json:"refreshInterval,omitempty"`
 }
+
+
+
+// Docker SD configurations allow retrieving scrape targets from Docker Engine hosts.
+// This SD discovers "containers" and will create a target for each network IP and port the container is configured to expose.
+type DockerSDConfig struct{
+	// ProxyConfig allows customizing the proxy behaviour for this scrape config.
+	// +optional
+	*v1.ProxyConfig `json:",inline"`
+	// TLS configuration applying to the target HTTP endpoint.
+	// +optional
+	TLSConfig *v1.SafeTLSConfig `json:"tlsConfig,omitempty"`
+	// The port to scrape metrics from.
+	// +optional
+	Port *int `json:"port,omitempty"`
+    // The host to use if the container is in host networking mode.
+    // +optional
+    HostNetworkingHost *string `json:"hostNetworkingHost,omitempty"`
+    // Optional filters to limit the discovery process to a subset of the available resources.
+    // +optional
+    Filters *map[string][]string `json:"filters,omitempty"`
```

Comment:
Wondering whether we should have a type like EC2Filter here
https://pkg.go.dev/github.com/docker/docker/api/types/filters#KeyValuePair

Reply:
I initially did make it a custom type, but then decided to not do that. Perhaps I should create a custom type. Curious, apart from possibly improved readability and consistency with the `KeyValuePair` you have linked to, does the change offer any other advantages?